builtin/logical/aws/path_static_roles.go

```diff
@@ -0,0 +1,300 @@
+}
+
+func pathStaticRoles(b *backend) *framework.Path {
+	roleResponse := map[int][]framework.Response{
+		http.StatusOK: {{
+			Description: http.StatusText(http.StatusOK),
+			Fields: map[string]*framework.FieldSchema{
+				paramRoleName: {
+					Type:        framework.TypeString,
+					Description: descRoleName,
+				},
+				paramUsername: {
+					Type:        framework.TypeString,
+					Description: descUsername,
+				},
+				paramRotationPeriod: {
+					Type:        framework.TypeDurationSecond,
+					Description: descRotationPeriod,
+				},
+			},
+		}},
+	}
+
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%s/%s", pathStaticRole, framework.GenericNameWithAtRegex(paramRoleName)),
+		Fields: map[string]*framework.FieldSchema{
+			paramRoleName: {
+				Type:        framework.TypeString,
+				Description: descRoleName,
+			},
+			paramUsername: {
+				Type:        framework.TypeString,
+				Description: descUsername,
+			},
+			paramRotationPeriod: {
+				Type:        framework.TypeDurationSecond,
+				Description: descRotationPeriod,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback:  b.pathStaticRolesRead,
+				Responses: roleResponse,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback:                    b.pathStaticRolesWrite,
+				ForwardPerformanceSecondary: true,
+				ForwardPerformanceStandby:   true,
+				Responses:                   roleResponse,
+			},
+			logical.DeleteOperation: &framework.PathOperation{
+				Callback:                    b.pathStaticRolesDelete,
+				ForwardPerformanceSecondary: true,
+				ForwardPerformanceStandby:   true,
+				Responses: map[int][]framework.Response{
+					http.StatusNoContent: {{
+						Description: http.StatusText(http.StatusNoContent),
+					}},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathStaticRolesHelpSyn,
+		HelpDescription: pathStaticRolesHelpDesc,
+	}
+}
+
+func (b *backend) pathStaticRolesRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName, ok := data.GetOk(paramRoleName)
+	if !ok {
+		return nil, fmt.Errorf("missing '%s' parameter", paramRoleName)
+	}
+
+	b.roleMutex.RLock()
+	defer b.roleMutex.RUnlock()
+
+	entry, err := req.Storage.Get(ctx, formatRoleStoragePath(roleName.(string)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read configuration for static role %q: %w", roleName, err)
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	var config staticRoleEntry
+	if err := entry.DecodeJSON(&config); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration for static role %q: %w", roleName, err)
+	}
+
+	return &logical.Response{
+		Data: formatResponse(config),
+	}, nil
+}
+
+func (b *backend) pathStaticRolesWrite(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
```

Comment:
This is not differentiating between a `CreateOperation` or an `UpdateOperation`. Usually we do that so that we don't require all the fields when we are doing an update. Is there a reason we decided not to do that here?

Reply:
I can't speak to the original design choice, but it does match (i think) what happens in the database backend with `func (b *databaseBackend) pathStaticRoleCreateUpdate()`

Reply:
Looks like that database backend method is doing different validation between create/update operations: https://github.com/hashicorp/vault/blob/0df02de1d576aa3f98931fd8f849c0d14e9a9b0b/builtin/logical/database/path_roles.go#L510-L530